mkdir_linux.go

```diff
@@ -0,0 +1,211 @@
+//go:build linux
+
+// Copyright (C) 2024 SUSE LLC. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package securejoin
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/sys/unix"
+)
+
+// partialOpen tries to lookup as much of the request path as possible and
+// opens the final existing component of the requested path, returning a file
+// handle to the final existing component and a string containing the remaining
+// path components (to be later created by MkdirAll).
+func partialOpen(root, unsafePath string) (_ *os.File, _ string, Err error) {
+	unsafePath = filepath.ToSlash(unsafePath) // noop
+
+	// Make sure there are no trailing slashes for the root path.
+	root = strings.TrimRight(root, "/")
```

Comment:
No, it was my attempt to avoid doing `path.Clean` but still making `checkProcSelfFdPath` work if you pass `"/foo/"` as a root path. Given the other comment, I suspect that I should've just copied the root "real" path caching we do in libpathrs.

I don't think you should use a symlink as the root, but I guess we shouldn't block it (also this means I need to drop `O_NOFOLLOW` from the root opening...).

And yeah, I missed the `"/"` case.